test/e2e/specs/editor/various/patterns.spec.js

```diff
@@ -149,28 +157,22 @@ test.describe( 'Synced pattern', () => {
 			.getByRole( 'button', { name: 'Create' } )
 			.click();
 
-		await expect
-			.poll(
-				editor.getBlocks,
-				'The block content should be wrapped by a pattern block wrapper'
-			)
-			.toEqual( [
-				{
-					name: 'core/block',
-					attributes: { ref: expect.any( Number ) },
-					innerBlocks: [
-						{
-							attributes: {
-								content: 'A useful paragraph to reuse',
-								dropCap: false,
-							},
-							innerBlocks: [],
-							name: 'core/paragraph',
-						},
-					],
-				},
-			] );
-		const after = await editor.getBlocks();
+		// Wait until the pattern is inserted.
+		await editor.canvas
+			.getByRole( 'document', {
+				name: 'Block: Pattern',
+			} )
+			.waitFor();
+
+		const [ syncedPattern ] = await editor.getBlocks();
+		const patternRecord = await getPatternRecord(
+			page,
+			syncedPattern?.attributes?.ref
+		);
+
+		expect( patternRecord?.content ).toBe(
+			'<!-- wp:paragraph -->\n<p>A useful paragraph to reuse</p>\n<!-- /wp:paragraph -->'
+		);
```

Comment:
I don't think this is necessary. We can just assert that the post content includes the `<!-- wp:block -->` block. Then we can visit the frontend to verify that the rendered post includes the pattern content (paragraph in this case.)

This kind of test only has to be done once and we only have to check that it contains `wp:block` in the remaining tests.

I think this also matches more closely to what the end users would expect. The tests are only what they seem because it's easier this way in the previous implementation, but they are technically testing implementation details.